src/Widgets/TerminalWidget.vala

```diff
@@ -519,46 +519,41 @@ namespace Terminal {
 
         protected override void paste_clipboard () {
             clipboard.request_text ((clipboard, text) => {
-                if (text == null) {
-                    return;
-                }
-
-                if (!text.validate ()) {
-                    warning ("Dropping invalid UTF-8 paste");
-                    return;
-                }
+                validated_feed (text);
+            });
+        }
 
+        // Check pasted and dropped text before feeding to child;
+        private void validated_feed (string? text) {
+            if (text != null && text.validate ()) {
+                var stripped_text = text.strip ();
                 unowned var toplevel = (MainWindow) get_toplevel ();
+                if (!toplevel.unsafe_ignored &&
+                    Application.settings.get_boolean ("unsafe-paste-alert")) {
 
-                if (!toplevel.unsafe_ignored && Application.settings.get_boolean ("unsafe-paste-alert")) {
                     string? warn_text = null;
-                    text._strip ();
-
                     if ("\n" in text) {
                         warn_text = _("The pasted text may contain multiple commands");
                     } else if ("sudo" in text || "doas" in text) {
                         warn_text = _("The pasted text may be trying to gain administrative access");
                     }
 
                     if (warn_text != null) {
-                        var dialog = new UnsafePasteDialog (toplevel, warn_text, text);
+                        var dialog = new UnsafePasteDialog (toplevel, warn_text, stripped_text);
                         dialog.response.connect ((res) => {
+                            dialog.destroy ();
                             if (res == Gtk.ResponseType.ACCEPT) {
-                               remember_command_start_position ();
-                               base.paste_clipboard ();
+                                feed_child (text.data);
                             }
-
-                            dialog.destroy ();
                         });
 
                         dialog.present ();
                         return;
```

Comment:
This return should be no longer necessary due to changes in this PR.

Reply:
Removed